cylc/flow/task_job_mgr.py

```diff
@@ -262,8 +262,8 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
 
         Return (list): list of tasks that attempted submission.
         """
-        if is_simulation:
-            return self._simulation_submit_task_jobs(itasks, workflow)
+        itasks, nonlive_tasks = self._nonlive_submit_task_jobs(
+            itasks, workflow, run_mode)
 
         # Prepare tasks for job submission
         prepared_tasks, bad_tasks = self.prep_submit_task_jobs(
```

Comment:
Some more of this method should be moved into the `submit_livelike_task_jobs` method (otherwise we can't mock out the live job preparation).

See the suggested diff in this comment: https://github.com/cylc/cylc-flow/pull/6039#discussion_r1765053779

Reply:
This isn't quite resolved yet, there's still the subshell eval in there as well as some `return` statements that I don't think make sense. See above suggestion.